tools/Directory.Build.props

```diff
@@ -0,0 +1,8 @@
+<Project>
+    <!-- To suppress inherited Directory.Build.props -->
+
+    <PropertyGroup>
+        <LangVersion>latest</LangVersion>
+    </PropertyGroup>
+
+</Project>
```

Comment:
Why are we suppressing the Directory.Build.props?

I could check out and remove this file and find out myself, I suppose.

Reply:
Hmm. Mostly things we could easily fix (make a few fields private and const) plus 

> CSC : warning CS8032: An instance of analyzer Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.ParameterNamesShouldMatchBaseDeclarationAnalyzer cannot be created from D:\Users\amidu\.nuget\packages\microsoft.codequality.analyzers\2.6.0\analyzers\dotnet\cs\Microsoft.CodeQuality.Analyzers.dll : Could not load file or assembly 'Microsoft.CodeAnalysis, Version=2.6.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.. [D:\Sandbox\FakeItEasy\tools\FakeItEasy.Build\FakeItEasy.Build.csproj]

This seems familiar, but I can't remember how to get around it. One option is to push 
```
<PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.6.0" PrivateAssets="all" />
```

down into src and tests' Directory.Build.props

Reply:
Well, the root Directory.Build.props has plenty of things we don't necessarily want in the build project: analyzers, rule sets, strong name key... So I thought it would be simpler to start from a blank slate.
And also the warnings you noticed. I don't know what's causing them. I can look into it if you like.

Reply:
I'm not against analyzers and rule sets being in the build project (admittedly they are less important).  Chasing the Microsoft.CodeQuality.Analyzers warnings is probably not worth it, so however you want to avoid, I'm content. Either as you have the directory.build.props as you do or maybe include the StyleCop analyzer (but I don't feel strongly enough to insist even on that; it's just that I'd default to attempting to preserver the style from the rest of the project).

Reply:
Apparently the issue is caused by the SDK version we use (2.0.3). I don't get those warnings with the latest (2.1.400). We should probably upgrade, but I get build errors in the FIE analyzer projects with the latest SDK, not sure why... For now, I guess we could just remove the `Microsoft.CodeAnalysis.FxCopAnalyzers` package from the build project, and solve the issue separately.